.github/workflows/main.yml

```diff
@@ -1,33 +1,40 @@
-# This is a basic workflow to help you get started with Actions
-
 name: CI
-
-# Controls when the action will run. Triggers the workflow on push or pull request
-# events but only for the master branch
 on:
   push:
-    branches: [master]
+    branches:
+      - master
+    tags:
+      - v*
   pull_request:
-    branches: [master]
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
-  build:
-    # The type of runner that the job will run on
-    runs-on: ubuntu-latest
+  modernize:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
+        os: [macOS-latest, ubuntu-latest, windows-latest]
 
-    # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
```

Comment:
@jreese do you think we should do pip caching?

https://github.com/actions/cache/blob/master/examples.md#using-pip-to-get-cache-location

Reply:
I've never worried about it in my other projects, and setting up the virtualenv has never seemed to be a bottleneck. I'm assuming they already do pretty heavy proxy caching already.

Reply:
seems fast enough tbh